### PR TITLE
fix(provider): stop dropping custom no-auth providers on restart

### LINKED ISSUE #555: Custom configured local provider is not persistent across runs
### Version / branch / commit

0.2.0

### OS and environment

Windows 11

### Steps to reproduce

1. Run Zero
2. Use /provider to setup a custom OpenAI provider. In the tested case: llama.cpp at local network address http://192.168.x.y:8080/v1 , no API key, custom name and model label
3. Add as Favorite (bonus check)
4. Use /model to select the newly created model 
5. quit Zero
6. run Zero
7. use /model to select the custom model created in step 2

### Expected behavior

The custom model is present and can be selected

### Actual behavior

The custom model is not listed anymore among the available models.

Bonus: the issue can be reproduced whether the custom model was marked as favorite or not.

### Relevant logs, screenshots, or error messages

_No response_

### PR DESCRIPTION
**Pull request by PierrunoYT**
## Summary

- Custom OpenAI/Anthropic-compatible catalog entries (`custom-openai-compatible`, `custom-anthropic-compatible`) hardcode `RequiresAuth: true` as a generic template default for the setup wizard's credential step — it isn't a fact about the user's actual endpoint.
- Leaving the credential step blank (e.g. pointing at a local, no-auth llama.cpp server) still stamped the saved profile with a guessed `OPENAI_API_KEY`/`ANTHROPIC_API_KEY` env var name.
- On the next run, that same static `RequiresAuth` flag made the profile look like it was missing a required credential, so it was silently filtered out of the usable-provider list and disappeared from `/model` — whether or not it was marked as a favorite.
- Added a `Custom` flag on the catalog `Descriptor` so both the write side (`providerWizardProfile`) and the read side (`setupMissingCredentialEnv`) judge these two catalog entries by the profile's own explicit `APIKeyEnv`, instead of treating the catalog's static `RequiresAuth` as authoritative.

Fixes #555

## Test plan

- [x] `go build ./...`
- [x] `go test ./internal/providercatalog/... ./internal/cli/... ./internal/tui/...` — all pass except one pre-existing, unrelated failure (`TestProviderWizardAdvancesProviderAPIKeyAndModelSteps`), confirmed to also fail identically on unmodified `main`.
- [x] Updated `TestProviderWizardCustomCompatibleProviderCollectsEndpointAndModel` to assert the corrected (no phantom env var) behavior.
- [x] Added regression cases to `TestSetupMissingCredentialEnv` covering: custom provider with no credential (now usable), custom provider with an explicit `APIKeyEnv` (still reported as missing until set), and the anthropic-compatible custom variant.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

## Summary by CodeRabbit

* **Bug Fixes**
  * Fixed missing-credential and auth-requirement detection for custom “*-compatible” provider endpoints, preventing stale legacy env vars from triggering errors.
  * Updated runtime readiness checks to consistently use a single source of truth for credential-env expectations.
* **New Features**
  * Added provider-catalog-aware credential-env resolution for saved provider profiles.
  * Marked custom “*-compatible” descriptors as explicitly custom to match UI and setup behavior.
* **Tests**
  * Expanded regression coverage for custom OpenAI-compatible and Anthropic-compatible profiles across multiple credential scenarios.
* **Documentation**
  * Clarified in the provider UI how custom endpoints handle authentication and environment-variable labeling.

### CODE REVIEW

File: internal/tui/provider_wizard.go [hunk 1]
```diff
@@ -1683,7 +1683,12 @@ func providerWizardProfile(provider providercatalog.Descriptor, model string, ap
 	}
 	if apiKey = strings.TrimSpace(apiKey); apiKey != "" {
 		profile.APIKey = apiKey
-	} else if env := firstProviderDisplayValue(provider.AuthEnvVars...); provider.RequiresAuth && env != "" {
+	} else if env := firstProviderDisplayValue(provider.AuthEnvVars...); provider.RequiresAuth && env != "" && !provider.Custom {
+		// Custom endpoints don't get a guessed env var default: RequiresAuth on
+		// these two catalog entries is just the wizard's template default (it
+		// doesn't know whether the user's own endpoint needs auth), so a blank
+		// credential step here means "this endpoint needs no auth," not "read
```
[the remark this answers is not in the document]
Comment (PierrunoYT) [reply internal/tui/provider_wizard.go:1841]: Fixed in 0d2170b: providerWizardCredentialLabel now takes the same !provider.Custom guard as providerWizardProfile, so the Done summary shows "not required" instead of "OPENAI_API_KEY env var" when the credential step is left blank for a custom endpoint — matching the profile that actually gets saved.

File: internal/providercatalog/catalog.go [hunk 2]
```diff
@@ -48,6 +48,14 @@ type Descriptor struct {
 	SupportedAPIFormats []APIFormat
 	Aliases             []string
 
+	// Custom marks the "bring your own endpoint" catalog entries
+	// (custom-openai-compatible, custom-anthropic-compatible). Unlike every other
+	// descriptor, RequiresAuth here is just a template default for the credential
+	// wizard step, not a fact about the user's actual endpoint — a custom target may
+	// or may not need auth, so callers deciding whether a *saved profile* is missing
+	// a credential must not treat RequiresAuth as authoritative for these entries.
+	Custom bool
```
[the remark this answers is not in the document]
Comment (PierrunoYT) [reply internal/providercatalog/catalog.go:57]: Good catch — confirmed the same descriptor.RequiresAuth-as-authoritative bug in validateProviderRuntimeReady (zero providers check would hard-error a no-auth custom endpoint) and providerCredentialRequired (/providers status would still warn). Fixed in 0d2170b:

- Added config.ProviderProfile.MissingCredentialEnv as the single source of truth (custom-endpoint handling + the legacy-default self-heal), and switched setupMissingCredentialEnv and validateProviderRuntimeReady to delegate to it.
- providerCredentialRequired (and providerWizardCredentialLabel) get the same Custom-aware check directly rather than delegating, since their catalog-less fallback path uses runtime-resolved provider-kind metadata (via the snapshot) that isn't safe to collapse into the profile-only canonical helper without changing behavior for that edge case.

Added direct unit tests for all three surfaces (config, cli, tui).

**Comment by PierrunoYT** [comment]
Addressing CodeRabbit's outside-diff finding on `internal/cli/setup.go:504-529` (legacy custom profiles needing migration):

Confirmed this was a real gap — a profile saved by the pre-fix wizard would already carry the phantom `APIKeyEnv` (e.g. `OPENAI_API_KEY`) it stamped on every custom provider, and the original fix would have kept treating that as an explicit requirement, leaving already-affected users' providers filtered out of `/model` even after upgrading.

Fixed in 375681d without a config.json migration: an `APIKeyEnv` that exactly matches the catalog's own guessed default is now treated the same as unset (self-heals silently on next read), while a deliberately different env var name (e.g. via `--api-key-env`) still means auth is required. Added a direct regression test for this in `setup_test.go`.

Also fixed two more surfaces Copilot's review caught with the same root-cause bug (`validateProviderRuntimeReady` / `zero providers check`, and the `/providers` status + wizard summary display) in 0d2170b, centralizing the shared logic into `config.ProviderProfile.MissingCredentialEnv` so these can't drift out of sync again.

**Review by Vasanthdev2004 — APPROVED** [review]
Clean fix for #555. The bug was that the wizard auto-stamped `APIKeyEnv` with the catalog's guessed default even for custom no-auth endpoints, so on restart `setupMissingCredentialEnv` reported the provider as missing a credential and it got filtered out of /model. Consolidating the three duplicated implementations (CLI setup, `providers check`, TUI status/wizard) into one `ProviderProfile.MissingCredentialEnv` source of truth is the right move — those surfaces can't drift again.

The custom-descriptor logic is the crux and it's handled well: for `custom-openai-compatible` / `custom-anthropic-compatible`, a credential is only expected when `APIKeyEnv` is set and differs from the catalog default. A value matching the default is indistinguishable from the old auto-stamp and self-heals pre-fix profiles, which means already-broken saves start working without a config rewrite. Regression tests cover the no-auth case, the explicit-env case, and the legacy-stale-default self-heal — that's the right matrix.

One tradeoff worth naming so it's not a surprise later: a custom endpoint that legitimately authenticates via the default-named env var (e.g. a proxy that reads `OPENAI_API_KEY`) now reports "not missing" and will surface in /model even when that env var is unset, so calls go out without auth and fail at the provider rather than being hidden up front. The escape hatch is an explicit `APIKey` in the profile or a non-default `APIKeyEnv`. Acceptable for a self-healing bug fix — the dropped-on-restart case is far more common — and the comment on `MissingCredentialEnv` already calls out the reasoning.

CI is green across the board. Approving; the issue-approved gate on #555 is in place.

**Comment by kevincodex1** [comment]
@PierrunoYT thanks for contribution please rebase to main anad fix conflicts

**Review by kevincodex1 — APPROVED** [review]
LGTM
